polaris-core/src/main/java/org/apache/polaris/core/auth/AuthenticatedPolarisPrincipal.java

```diff
@@ -56,10 +72,20 @@ public List<PrincipalRoleEntity> getActivatedPrincipalRoles() {
     return activatedPrincipalRoles;
   }
 
+  /** FIXME this method makes this class mutable, which seems risky */
   public void setActivatedPrincipalRoles(List<PrincipalRoleEntity> activatedPrincipalRoles) {
     this.activatedPrincipalRoles = activatedPrincipalRoles;
   }
 
+  /**
+   * Whether all roles should be activated. This is true when the principal is requesting all
+   * available principal roles. When this is true, {@link #getActivatedPrincipalRoleNames()} returns
+   * an empty set.
+   */
+  public boolean allRoles() {
```

Comment:
I think we need this method to distinguish the case when the principal is requesting all roles from the case where no roles were found in the token – since both cases translate into an empty set of role names.

Reply:
But a better solution imho would be to remove roles from this class completely, and let `SecurityIdentity.getRoles()` expose the roles.

I checked the call sites for `getActivatedPrincipalRoles()` and it's in fact never used. So it seems that `setActivatedPrincipalRoles()` could be removed as well.

Reply:
I removed the method because it doesn't work well with the `Resolver`. TLDR we can't distinguish "no roles" from "all roles" for now.